extensions-core/stats/src/main/java/io/druid/query/aggregation/teststats/PvaluefromZscorePostAggregator.java

```diff
@@ -0,0 +1,141 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.teststats;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Sets;
+import io.druid.query.Queries;
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.query.aggregation.PostAggregator;
+import io.druid.query.aggregation.post.ArithmeticPostAggregator;
+import io.druid.query.aggregation.post.PostAggregatorIds;
+import io.druid.query.cache.CacheKeyBuilder;
+import org.apache.commons.math3.distribution.NormalDistribution;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Set;
+
+@JsonTypeName("pvalue2tailedZtest")
+public class PvaluefromZscorePostAggregator implements PostAggregator
+{
+  private final String name;
+  private final PostAggregator zScore;
+
+  @JsonCreator
+  public PvaluefromZscorePostAggregator(
+      @JsonProperty("name") String name,
+      @JsonProperty("zScore") PostAggregator zScore
+  )
+  {
+    Preconditions.checkNotNull(
+        name,
+        "Must have a valid, non-null post-aggregator"
+    );
+    Preconditions.checkNotNull(
+        zScore,
+        "Must have a valid, non-null post-aggregator"
+    );
+    this.name = name;
+    this.zScore = zScore;
+  }
+
+  @Override
+  public Set<String> getDependentFields()
+  {
+    Set<String> dependentFields = Sets.newHashSet();
+
+    dependentFields.addAll(zScore.getDependentFields());
+
+    return dependentFields;
+  }
+
+  @Override
+  public Comparator getComparator()
+  {
+    return ArithmeticPostAggregator.DEFAULT_COMPARATOR;
+  }
+
+  @Override
+  public Object compute(Map<String, Object> combinedAggregators)
+  {
+
+    double zScoreValue = ((Number) zScore.compute(combinedAggregators))
+        .doubleValue();
+
+    zScoreValue = Math.abs(zScoreValue);
+    return 2 * (1 - cumulativeProbability(zScoreValue));
+  }
+
+  private double cumulativeProbability(double x)
+  {
+    try {
+      NormalDistribution normDist = new NormalDistribution();
+      return normDist.cumulativeProbability(x);
+    }
+    catch (IllegalArgumentException ex) {
+      return Double.NaN;
+    }
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  @Override
+  public PostAggregator decorate(Map<String, AggregatorFactory> aggregators)
+  {
+    return new PvaluefromZscorePostAggregator(
+        name,
+        Iterables.getOnlyElement(Queries.decoratePostAggregators(
+            Collections.singletonList(zScore), aggregators))
+    );
+  }
+
+  @Override
+  public int hashCode()
```

Comment:
Please implement equals() method either. This will remove the potential problem when working with collections.